src/Microsoft.AspNetCore.Razor.Runtime/Runtime/TagHelpers/TagHelperDescriptorFactory.cs

```diff
@@ -750,5 +733,311 @@ private static string ToHtmlCase(string name)
+            {
+                Debug.Assert(At('['));
+
+                // Move past '['.
+                Next();
+                PassOptionalWhitespace();
+
+                var attributeName = ParseCSSAttributeName(errorSink);
+
+                if (attributeName == null ||
+                    !ValidateName(attributeName, targetingAttributes: true, errorSink: errorSink))
+                {
+                    // Couldn't parse a valid attribute name.
+                    return null;
+                }
+
+                PassOptionalWhitespace();
+
+                if (!EnsureNotAtEnd(errorSink))
+                {
+                    return null;
+                }
+
+                var valueOperator = ParseCSSValueOperator(errorSink);
+
+                PassOptionalWhitespace();
+
+                if (!EnsureNotAtEnd(errorSink))
+                {
+                    return null;
+                }
+
+                var value = ParseCSSValue(errorSink);
+
+                if (value == null)
+                {
+                    // Couldn't parse value
+                    return null;
+                }
+
+                PassOptionalWhitespace();
+
+                if (At(']'))
+                {
+                    // Move past the ending bracket.
+                    Next();
+                }
+                else
+                {
+                    errorSink.OnError(
+                        SourceLocation.Zero,
+                        Resources.TagHelperDescriptorFactory_CouldNotFindMatchingEndBrace,
+                        length: 0);
+                    return null;
+                }
+
+                return new TagHelperRequiredAttributeDescriptor
+                {
+                    Name = attributeName,
+                    Value = value,
+                    Operator = valueOperator,
+                    IsCssSelector = true,
+                };
+            }
+
+            private bool EnsureNotAtEnd(ErrorSink errorSink)
+            {
+                if (AtEnd)
+                {
+                    errorSink.OnError(
+                        SourceLocation.Zero,
+                        Resources.TagHelperDescriptorFactory_CouldNotFindMatchingEndBrace,
+                        length: 0);
+
+                    return false;
+                }
+
+                return true;
+            }
+
+            private void Next()
+            {
+                _index = Math.Min(_index + 1, _requiredAttributes.Length);
+            }
+
+            private bool NextIs(char c)
+            {
+                _index++;
+                var nextIs = At(c);
+                _index--;
+
+                return nextIs;
+            }
+
+            private bool At(char c)
+            {
+                return !AtEnd && Current == c;
+            }
+
+            private void PassOptionalWhitespace()
+            {
+                while (!AtEnd && char.IsWhiteSpace(Current))
```

Comment:
Just use `' '` and `'\t'` since that's all we're supporting elsewhere. Strangely inconsistent otherwise.